File: solutions/_977_squares_of_a_sorted_array/solution.java
package _977_squares_of_a_sorted_array;

class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] result = new int[nums.length];
        int L = 0;
        int R = nums.length - 1;

        while (L <= R) {
            int left = nums[L];
            int right = nums[R];
            int idx = R - L;
            if (Math.abs(left) > Math.abs(right)) {
                result[idx] = left * left;
                L++;
            } else {
                result[idx] = right * right;
                R--;
            }
        }
        return result;
    }
}

/*
 * https://leetcode.com/problems/squares-of-a-sorted-array/
 */

File: solutions/_977_squares_of_a_sorted_array/solution.py
from typing import List


class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0] * n
        L = 0
        R = n - 1

        while L <= R:
            left = abs(nums[L])
            right = abs(nums[R])
            if left > right:
                res[R - L] = left**2
                L += 1
            else:
                res[R - L] = right**2
                R -= 1

        return res


# https://leetcode.com/problems/squares-of-a-sorted-array/
